nixos/modules/services/networking/netmaker/netmaker.nix

```diff
@@ -0,0 +1,787 @@
+      }];
+    }
+    {
+      systemd.tmpfiles.rules = [
+        "d ${cfg.secretsDir} 1700 root root"
+        "f ${envFile.generated.config} 1644 root root"
+        "f ${envFile.generated.secrets} 1600 root root"
+        "f ${envFile.user.config} 1644 root root"
+        "f ${envFile.user.secrets} 1600 root root"
+      ];
+
+      services.netmaker.environmentFiles = [
+        envFile.generated.config
+        envFile.generated.secrets
+        envFile.user.config
+        envFile.user.secrets
+        (lib.mkAfter cfg.environment)
+      ];
+
+      services.netmaker.config = {
+        server.apiconn = "${cfg.api.domain}:443";
+        server.apihost = cfg.api.domain;
+        server.apilistenaddress = cfg.api.internal.host;
+        server.apiport = builtins.toString cfg.api.internal.port;
+        server.broker = "wss://${cfg.mqtt.domain}";
+        server.brokertype = cfg.mqtt.type;
+        server.database = cfg.db.type;
+        server.frontendurl = "https://${cfg.ui.domain}";
+        server.jwt_validity_seconds = cfg.jwtValiditySeconds;
+        server.mqusername = cfg.mqtt.username;
+        server.server = cfg.domain;
+        server.serverbrokerendpoint = "ws://${cfg.mqtt.internal.addr}";
+        server.telemetry = if cfg.telemetry then "on" else "off";
+        server.verbosity = cfg.verbosity;
+        # don't set to anything, see https://github.com/nazarewk/netmaker/blob/9ca6b44228847d246dd5617b73f69ec26778f396/servercfg/serverconf.go#L215-L223
+        server.corednsaddr = "";
+      };
+
+      systemd.services.netmaker-configure = {
+        serviceConfig.Type = "oneshot";
+        serviceConfig.RemainAfterExit = true;
+        script = lib.getExe (pkgs.writeShellApplication {
+          name = "netmaker-generate-config";
+          runtimeInputs = with pkgs; [
+            coreutils
+            gawk
+            gnugrep
+            gnused
+            systemd
+          ];
+          text = ''
+            MQ_PASSWORD_FILE="${cfg.mqtt.passwordFile}"
+
+            SECRETS="${envFile.generated.secrets}"
+            ENV="${envFile.generated.config}"
+
+            ${builtins.readFile ./netmaker-generate-config.sh}
+          '';
+        });
+      };
+
+      # based on https://github.com/gravitl/netclient/blob/b9ea9c9841f01297955b03c2b5bbf4b5139aa40c/daemon/systemd_linux.go#L14-L30
+      systemd.services.netmaker = {
+        description = "Netmaker Server daemon";
+        after = [ "network-online.target" "netmaker-configure.service" ];
+        wants = [ "network-online.target" ];
+        requires = [ "netmaker-configure.service" ];
+        wantedBy = [ "multi-user.target" ];
+
+        serviceConfig.EnvironmentFile = cfg.environmentFiles;
+        serviceConfig.WorkingDirectory = cfg.dataDir;
+        serviceConfig.ExecStart = "${lib.getExe cfg.package} -c ${cfg.config}";
+        serviceConfig.ExecStartPost = pkgs.writeShellScript "configure-nmctl" ''
+          export HOME=/root
```

Comment:
Should the service really dump it's files into the root users' home?

Reply:
I don't like it either, but people might expect `nmctl` to work out of the box, because it is part of the default non-NixOS installation https://github.com/gravitl/netmaker/blob/b9563fbff45dcd1fef459fc23a79b7ea6dea7cbe/scripts/nm-quick.sh#L172-L193